RemoteInput/Plugin/ControlCenter.cxx

```diff
@@ -1089,6 +1125,206 @@ void ControlCenter::process_reflect_array_with_size(jarray array, void* &argumen
+				//reflector->getEnv()->GetCharArrayRegion(static_cast<jcharArray>(array), index, length, static_cast<jchar*>(response));
+			}
+			break;
+
+			case ReflectionArrayType::BYTE:
+			{
+				jbyte* result = reflector->getEnv()->GetByteArrayElements(static_cast<jbyteArray>(array), nullptr);
+				for (std::size_t i = 0; i < length; ++i)
+				{
+					//fprintf(stderr, "write byte\n");
+					EIOS_Write(response, ReflectionBinaryType::BYTE);
+					//fprintf(stderr, "write byte[%zu]\n", i);
+					EIOS_Write(response, result[i]);
+				}
+				reflector->getEnv()->ReleaseByteArrayElements(static_cast<jbyteArray>(array), result, JNI_ABORT);
+				//reflector->getEnv()->GetByteArrayRegion(static_cast<jbyteArray>(array), index, length, static_cast<jbyte*>(response));
+			}
+			break;
+
+			case ReflectionArrayType::BOOL:
+			{
+				jboolean* result = reflector->getEnv()->GetBooleanArrayElements(static_cast<jbooleanArray>(array), nullptr);
+				for (std::size_t i = 0; i < length; ++i)
+				{
+					//fprintf(stderr, "write bool\n");
+					EIOS_Write(response, ReflectionBinaryType::BOOL);
+					//fprintf(stderr, "write bool[%zu]\n", i);
+					EIOS_Write(response, result[i]);
+				}
+				reflector->getEnv()->ReleaseBooleanArrayElements(static_cast<jbooleanArray>(array), result, JNI_ABORT);
+				//reflector->getEnv()->GetBooleanArrayRegion(static_cast<jbooleanArray>(array), index, length, static_cast<jboolean*>(response));
+			}
+			break;
+
+			case ReflectionArrayType::SHORT:
+			{
+				jshort* result = reflector->getEnv()->GetShortArrayElements(static_cast<jshortArray>(array), nullptr);
+				for (std::size_t i = 0; i < length; ++i)
+				{
+					//fprintf(stderr, "write short\n");
+					EIOS_Write(response, ReflectionBinaryType::SHORT);
+					//fprintf(stderr, "write short[%zu]\n", i);
+					EIOS_Write(response, result[i]);
+				}
+				reflector->getEnv()->ReleaseShortArrayElements(static_cast<jshortArray>(array), result, JNI_ABORT);
+				//reflector->getEnv()->GetShortArrayRegion(static_cast<jshortArray>(array), index, length, static_cast<jshort*>(response));
+			}
+			break;
+
+			case ReflectionArrayType::INT:
+			{
+				jint* result = reflector->getEnv()->GetIntArrayElements(static_cast<jintArray>(array), nullptr);
+				for (std::size_t i = 0; i < length; ++i)
+				{
+					//fprintf(stderr, "write int\n");
+					EIOS_Write(response, ReflectionBinaryType::INT);
+					//fprintf(stdout, "write int[%zu]\n", i);
+					EIOS_Write(response, result[i]);
+				}
+				reflector->getEnv()->ReleaseIntArrayElements(static_cast<jintArray>(array), result, JNI_ABORT);
+				//reflector->getEnv()->GetIntArrayRegion(static_cast<jintArray>(array), index, length, static_cast<jint*>(response));
+			}
+			break;
+
+			case ReflectionArrayType::LONG:
+			{
+				jlong* result = reflector->getEnv()->GetLongArrayElements(static_cast<jlongArray>(array), nullptr);
+				for (std::size_t i = 0; i < length; ++i)
```

Comment:
Why do we need to write the type? If we know the entire array is `LONG` based on `GetLongArrayElements`, we should just write the whole array with `GetLongArrayRegion` instead of looping over it and writing the type plus the value since we already know the type?

Reply:
My brain was melting without having debug logs so I switched to item-by-item processing. I'll revisit now that debug is working for me.